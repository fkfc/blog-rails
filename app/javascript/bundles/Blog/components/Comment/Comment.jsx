import React from 'react';
import PropTypes from 'prop-types';

export default function Comment(props) {
  const {
    id, content, createdAt, postCommentPath,
  } = props;

  if (typeof id === 'undefined' || id === null) return null;
  const editPostCommentPath = `${postCommentPath}/edit`;
  return (
    <li className="list-group-item comment">
      <pre>
        { content }
      </pre>

      <div className="boxFooter">
        <div className="text">
          { createdAt }
        </div>
        <div className="btn-group btn-group-sm" role="group">
          <a href={editPostCommentPath} className="btn btn-secondary">Edit</a>
          <a href={postCommentPath} className="btn btn-secondary" rel="nofollow" data-method="delete">Remove</a>
        </div>
      </div>

    </li>
  );
}

Comment.propTypes = {
  id: PropTypes.number,
  content: PropTypes.string,
  createdAt: PropTypes.string,
  postCommentPath: PropTypes.string.isRequired,
};

Comment.defaultProps = {
  id: null,
  content: '',
  createdAt: null,
};
