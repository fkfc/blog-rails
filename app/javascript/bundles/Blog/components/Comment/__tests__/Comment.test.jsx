import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Comment from '../Comment';

describe('Testing the Post Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      id: 1,
      content: 'comment content',
      createdAt: 'Aaa, 12 3456',
      postCommentPath: '/postCommentPath',
    };
  });
  it('Should render correctly', () => {
    const wrapper = shallow(<Comment {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('Includes the comment content', () => {
    const wrapper = shallow(<Comment {...props} />);
    expect(wrapper.text()).toContain(props.content);
  });
  it('Includes the creation date', () => {
    const wrapper = shallow(<Comment {...props} />);
    expect(wrapper.text()).toContain(props.createdAt);
  });
  it('Renders nothing when it doesnt have an ID', () => {
    props.id = null;
    const wrapper = shallow(<Comment {...props} />);
    expect(wrapper.type()).toBe(null);
  });
});
