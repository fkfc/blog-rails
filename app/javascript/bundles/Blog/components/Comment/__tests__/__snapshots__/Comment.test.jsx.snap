// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Post Component Should render correctly 1`] = `
<li
  className="list-group-item comment"
>
  <pre>
    comment content
  </pre>
  <div
    className="boxFooter"
  >
    <div
      className="text"
    >
      Aaa, 12 3456
    </div>
    <div
      className="btn-group btn-group-sm"
      role="group"
    >
      <a
        className="btn btn-secondary"
        href="/postCommentPath/edit"
      >
        Edit
      </a>
      <a
        className="btn btn-secondary"
        data-method="delete"
        href="/postCommentPath"
        rel="nofollow"
      >
        Remove
      </a>
    </div>
  </div>
</li>
`;
